### ISSUE TITLE
Upgrading of tied packages when one is in arch repos

### ISSUE THREAD

**Issue by ArchangeGabriel** [archlinuxfr/yaourt#308]
I have the following case:
* package A and package B are released together by upstream, and B needs A version number to match (so package B version e.g. 0.9.3 depends on `A=0.9.3`).
* package A is in the repo, but package B is in AUR.
* both packages get updated. When running `yaourt -Syua`, package A can’t be upgraded because it breaks package B dependency. When trying to upgrade package B first, it fails because installing package A upgrade breaks currently installed package A dependency.

Currently, the solution is to download remove package B, upgrade package A, reinstall package B but this is far less than ideal.

Not sure how to fix this though.

**Comment by f2404** [comment]
Does not `yaourt -S B` fix this? Or `yaourt -S B A`?

**Comment by ArchangeGabriel** [comment]
No. Like I’ve said above, upgrading package B first (`yaourt -S B`) results in not being able to create package B because it requires upgrading package A first (because of the specified version in the `depends`), which breaks currently installed package B, thus pacman refuse to install A. `yaourt -S B A` results in the same issue as upgrading A first, because yaourt try to upgrade A in the first place (or does that depends on other factors?).

**Comment by f2404** [comment]
Perhaps you could use `--nodeps` option to upgrade A, and then upgrade B.

**Comment by ArchangeGabriel** [comment]
In my case, that should work, provided they are no new deps. But what if they are some?

And finally, it wouldn’t work in the case the upgraded A package is actually needed to build the upgraded B package.

Maybe the `=x.y.z` part of the dependency should be dropped in AUR, since we might say this is no different than a required rebuild because of ABI change in any library depended on.

**Comment by f2404** [comment]
> And finally, it wouldn’t work in the case the upgraded A package is actually needed to build the
> upgraded B package.

Why? Package B will be built from AUR after A is upgraded, right?

> Maybe the =x.y.z part of the dependency should be dropped in AUR, since we might say this is no
> different than a required rebuild because of ABI change in any library depended on.

Yes, I think this issue should likely be resolved in package B PKGBUILD; I don't see any automated way of resolving it during package upgrade which couldn't potentially lead to a broken system.

**Comment by ArchangeGabriel** [comment]
Oh, sorry, I misread your comment and thought you meant upgrading B with `--nodeps`.

Indeed, upgrading only A with this option should solve this issue. But I’ll go the other way (changing PKGBUILD).

**Comment by f2404** [comment]
I'm afraid this issue cannot be solved by yaourt in a common and fail-proof way.
